docs/docs/recipes/working-with-plugins.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Recipes: Working with Plugins"
+tableOfContentsDepth: 1
+---
+
+A [Gatsby plugin](/docs/what-is-a-plugin/) abstracts Gatsby APIs into an installable package. This means that modular chunks of Gatsby functionality isn’t directly written into your project, but rather versioned, centrally managed, and installed as a dependency. You can add external data, transform data, add third-party services (e.g. Google Analytics, Stripe), and more.
+
+## Using a plugin
+
+Found a plugin you'd like to use on your project? Awesome! You can configure it for use by following the steps below. This recipe uses the [`gatsby-source-filesystem` plugin](/plugins/gatsby-source-filesystem) as the example because it is one of the most popular commonly used plugins.
+
+> If you'd like to take a look at the available plugins you can install, check out the [plugin library](/plugins).
+
+### Prerequisites
+
+- An existing [Gatsby site](/docs/quick-start/) with a `gatsby-config.js` file
+
+### Directions
+
+1. Install the `gatsby-source-filesystem` plugin by running the following command:
+
+```shell
+npm install gatsby-source-filesystem
+```
+
+2. Add the plugin to your `gatsby.config.js`, and set any options it needs, the filesystem source plugin takes a `name` and `path` as options:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    // highlight-start
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: `images`,
+        path: `${__dirname}/src/images`,
+      },
+    },
+    // highlight-end
+  ],
+}
+```
+
+_The instructions found in the README of the plugin you're using can you help you determine specifics about what configurations (if any) it requires. For this example to have an effect in your own site, you would need a folder of files at `src/images`._
```

Comment:
```suggestion
_The instructions found in the README of the plugin you're using can you help you determine specifics about what configurations (if any) it requires. For this example to have an effect in your own site, you would need to create the `src/images` folder and place files inside of it._
```

Reply:
Is it worth noting that not all plugins have options? Or no because this is a recipe and specific example?

Reply:
I suppose I sort of hinted at that:

> what configurations (if any) it requires

That still might not be super clear though 🤔

Reply:
Ah, that's a good point. Maybe instead of "if any" there could be a note call out mentioning that not all plugins have them? The if any might also be enough, I just missed it.